test/list.go

```diff
@@ -486,6 +488,75 @@ func DescribeListTestsFor(ctx *common.TestContext, t TestCase, responseMode Resp
 				})
 			}
 
+			if t.SupportsAsyncOperations {
+				Context("when attach_last_operations is truthy", func() {
+					It("retrieves the resources list when each resource contains the last operation it is associated to", func() {
+						resp := ctx.SMWithOAuth.GET(t.API).
+							WithQuery("attach_last_operations", "true").
+							Expect().Status(http.StatusOK).
+							JSON()
+						for _, resource := range resp.Path("$.items[*]").Array().Iter() {
+							currentResourceId := resource.Object().Value("id").String().Raw()
+							currentResourceState := resource.Object().Value("ready").Boolean().Raw()
+							lastOp := resource.Object().Value("last_operation")
+
+							Expect(lastOp.Object().Value("resource_id").String().Raw()).To(Equal(currentResourceId))
+							Expect(lastOp.Object().Value("ready").Boolean().Raw()).To(Equal(currentResourceState))
+							Expect(lastOp.Object().Value("state").String().Raw()).To(Equal("succeeded"))
+							Expect(lastOp.Object().Value("resource_type").String().Raw()).ToNot(BeEmpty())
+							Expect(lastOp.Object().Value("type").String().Raw()).ToNot(BeEmpty())
+							Expect(lastOp.Object().Value("deletion_scheduled").String().Raw()).ToNot(BeEmpty())
+						}
+					})
+				})
+
+				Context("when the last operation does not exist", func() {
+					It("should return the resource without it", func() {
+						resp := ctx.SMWithOAuth.GET(t.API).
+							WithQuery("attach_last_operations", "true").
+							Expect().Status(http.StatusOK).
+							JSON()
+
+						resource := resp.Path("$.items[0]")
+						lastOp := resource.Object().Value("last_operation")
+						resourceId := lastOp.Object().Value("resource_id").String().Raw()
+						criteria := query.ByField(query.EqualsOperator, "resource_id", resourceId)
+						err := ctx.SMRepository.Delete(context.Background(), types.OperationType, criteria)
+						Expect(err).ShouldNot(HaveOccurred())
+
+						resp = ctx.SMWithOAuth.GET(t.API).
+							WithQuery("attach_last_operations", "true").
+							Expect().Status(http.StatusOK).
+							JSON()
+
+						var foundResource bool
+						for _, resource := range resp.Path("$.items[*]").Array().Iter() {
+							itemResourceId := resource.Object().Value("id").String().Raw()
+							if itemResourceId == resourceId {
+								foundResource = true
+								resource.Object().NotContainsKey("last_operation")
+							} else {
+								resource.Object().ContainsKey("last_operation")
+							}
+						}
+
+						Expect(foundResource).To(Equal(true))
+					})
+				})
+			}
+
+			Context("when attach_last_operations is falsy", func() {
```

Comment:
this should be in the if.SupportsAsync, otherwise it doesn't test much -  there are no operations anyway, right?